### ISSUE TITLE
Incorrect Insufficient Balance error in zkApp transaction

### ISSUE THREAD

**Issue by asimaranov** [aurowallet/auro-wallet-browser-extension#32]
When I try to make a transaction with tokens transfer, wallet shows unsufficient balance and wallet logs show that there's a failed request. Request:
<img width="1285" alt="image" src="https://github.com/aurowallet/auro-wallet-browser-extension/assets/25568730/e212eb02-b3da-4fa1-ad67-fbff7f2476a9">

Error:

<img width="1285" alt="image" src="https://github.com/aurowallet/auro-wallet-browser-extension/assets/25568730/62596f73-1ad7-4075-bb20-f1d884a7d656">

Transactions: 
```
{
  "feePayer": {
    "body": {
      "publicKey": "B62qiTKpEPjGTSHZrtM8uXiKgn8So916pLmNJKDhKeyBQL9TDb3nvBG",
      "fee": "0",
      "validUntil": null,
      "nonce": "0"
    },
    "authorization": "7mWxjLYgbJUkZNcGouvhVj5tJ8yu9hoexb9ntvPK8t5LHqzmrL6QJjjKtf5SgmxB4QWkDw7qoMMbbNGtHVpsbJHPyTy2EzRQ"
  },
  "accountUpdates": [
    {
      "body": {
        "publicKey": "B62qkh5QbigkTTXF464h5k6GW76SHL7wejUbKxKy5vZ9qr9dEcowe6G",
        "tokenId": "wSHV2S4qX9jFsLjQo8r1BsMLH2ZRKsZx6EJd1sbozGPieEC4Jf",
        "update": {
          "appState": [
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null
          ],
          "delegate": null,
          "verificationKey": null,
          "permissions": null,
          "zkappUri": null,
          "tokenSymbol": null,
          "timing": null,
          "votingFor": null
        },
        "balanceChange": {
          "magnitude": "10000000000",
          "sgn": "Negative"
        },
        "incrementNonce": true,
        "events": [],
        "actions": [],
        "callData": "0",
        "callDepth": 0,
        "preconditions": {
          "network": {
            "snarkedLedgerHash": null,
            "blockchainLength": null,
            "minWindowDensity": null,
            "totalCurrency": null,
            "globalSlotSinceGenesis": null,
            "stakingEpochData": {
              "ledger": {
                "hash": null,
                "totalCurrency": null
              },
              "seed": null,
              "startCheckpoint": null,
              "lockCheckpoint": null,
              "epochLength": null
            },
            "nextEpochData": {
              "ledger": {
                "hash": null,
                "totalCurrency": null
              },
              "seed": null,
              "startCheckpoint": null,
              "lockCheckpoint": null,
              "epochLength": null
            }
          },
          "account": {
            "balance": null,
            "nonce": {
              "lower": "0",
              "upper": "0"
            },
            "receiptChainHash": null,
            "delegate": null,
            "state": [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null
            ],
            "actionState": null,
            "provedState": null,
            "isNew": null
          },
          "validWhile": null
        },
        "useFullCommitment": false,
        "implicitAccountCreationFee": false,
        "mayUseToken": {
          "parentsOwnToken": false,
          "inheritFromParent": false
        },
        "authorizationKind": {
          "isSigned": true,
          "isProved": false,
          "verificationKeyHash": "3392518251768960475377392625298437850623664973002200885669375116181514017494"
        }
      },
      "authorization": {
        "proof": null,
        "signature": null
      }
    },
    {
      "body": {
        "publicKey": "B62qkh5QbigkTTXF464h5k6GW76SHL7wejUbKxKy5vZ9qr9dEcowe6G",
        "tokenId": "wSHV2S4qX9jFsLjQo8r1BsMLH2ZRKsZx6EJd1sbozGPieEC4Jf",
        "update": {
          "appState": [
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null
          ],
          "delegate": null,
          "verificationKey": null,
          "permissions": null,
          "zkappUri": null,
          "tokenSymbol": null,
          "timing": null,
          "votingFor": null
        },
        "balanceChange": {
          "magnitude": "10000000000",
          "sgn": "Positive"
        },
        "incrementNonce": false,
        "events": [],
        "actions": [],
        "callData": "0",
        "callDepth": 1,
        "preconditions": {
          "network": {
            "snarkedLedgerHash": null,
            "blockchainLength": null,
            "minWindowDensity": null,
            "totalCurrency": null,
            "globalSlotSinceGenesis": null,
            "stakingEpochData": {
              "ledger": {
                "hash": null,
                "totalCurrency": null
              },
              "seed": null,
              "startCheckpoint": null,
              "lockCheckpoint": null,
              "epochLength": null
            },
            "nextEpochData": {
              "ledger": {
                "hash": null,
                "totalCurrency": null
              },
              "seed": null,
              "startCheckpoint": null,
              "lockCheckpoint": null,
              "epochLength": null
            }
          },
          "account": {
            "balance": null,
            "nonce": null,
            "receiptChainHash": null,
            "delegate": null,
            "state": [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null
            ],
            "actionState": null,
            "provedState": null,
            "isNew": null
          },
          "validWhile": null
        },
        "useFullCommitment": false,
        "implicitAccountCreationFee": false,
        "mayUseToken": {
          "parentsOwnToken": false,
          "inheritFromParent": false
        },
        "authorizationKind": {
          "isSigned": false,
          "isProved": false,
          "verificationKeyHash": "3392518251768960475377392625298437850623664973002200885669375116181514017494"
        }
      },
      "authorization": {
        "proof": null,
        "signature": null
      }
    }
  ],
  "memo": "E4YM2vTHhWEg66xpj52JErHUBU4pZ1yageL4TVDDpTTSsv8mK6YaH"
}
```

**Comment by asimaranov** [comment]
Transaction:
![telegram-cloud-photo-size-2-5307567400549472344-y](https://github.com/aurowallet/auro-wallet-browser-extension/assets/25568730/5e619c1b-db3c-4759-a542-d1569229f4fc)

**Comment by asimaranov** [comment]
<img width="379" alt="image" src="https://github.com/aurowallet/auro-wallet-browser-extension/assets/25568730/e3a9a044-2382-4f83-83c5-cd66ecd51ccc">

How it displayed

**Comment by lvshaoping007** [comment]
Okay, we have successfully reproduced this problem and will soon release a version to fix this problem.

**Comment by lvshaoping007** [comment]
Hello, the problem has been fixed in the latest version [2.2.12](https://chromewebstore.google.com/detail/auro-wallet/cnmamaachppnkjgnildpdmkaakejnhae?hl=zh-CN&utm_source=ext_sidebar)
